gnome-initial-setup/eos-demo-mode

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+# eos-setup-demo-user - configure demo user account
+# Copyright (C) 2016-2017 Endless Mobile, Ltd.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import argparse
+import glob
+import json
+import logging
+import os
+import subprocess
+import sys
+
+from gi.repository import GLib, Gio
+
+log = logging.getLogger(sys.argv[0])
+
+ICON_GRID_DIR = '/var/lib/eos-image-defaults/icon-grid'
+
+DESKTOP_GRID_ID = 'desktop'
+
+GOOGLE_CHROME = 'google-chrome.desktop'
+CHROMIUM_BROWSER = 'chromium-browser.desktop'
+
+GDM_DAEMON_CONF = '/etc/gdm3/daemon.conf'
+
+
+def dq(s):
+    return '"{}"'.format(s)
+
+
+def remove_chrome_from_icon_grids():
```

Comment:
Per my discussion with this with @wjt on slack (tl;dr : the gsetting starts out empty and the shell always reads from the hardcoded icon grids as a fallback):

    So I’m looking at migrating all of eos-setup-live-user to just change settings from within g-i-s instead (since we have gnome-initial-setup-copy-worker which will just copy across settings changes for us)


    [3:45 PM] 
    But in e-s-l-u we alter the icon grids in /usr - is there any reason why we do that?


    [3:52 PM] 
    wjt To add the eos-installer.desktop to the desktop, and swap Chrome out and Chromium in


    [3:52 PM] 
    The former is self explanatory


    [3:53 PM] 
    The latter because we don't want live users to download chrome (it'll be discarded anyway, and installing it fills up most of the tmpfs) (edited)


    [3:55 PM] 
    Why do them system wide rather than via gsettings? To avoid reimplementing the shells logic for picking the correct grid for the locale .


    [3:55 PM] 
    (the gsettings default for the grid is empty string.) (edited)

    smspillaz @wjt I’m guessing the reason why we can’t just copy the existing value and s/google-chrome/chromium-browser/ is because the locale might not be set up as how the user wants to log in?


    [4:05 PM] 
    (in the gsetting, that is)


    [4:05 PM] 
    Reason I ask is because if we can do it entirely on the gsettings side, then we can completely get rid of `eos-setup-live-user` and `eos-setup-demo-user` and move all of that logic to g-i-s

    smspillaz
    though now that I think of it … perhaps it wouldn’t work because as you said, the setting starts out empty ….

    wjt right. the setup shell profile does not have an icon grid so I would not expect the setting to be populated. even if it were, it would be populated based on the default locale for the image, which may not match the locale the user picks in the FBE